File: Programmers/Lv.2/카펫/42842.js
function solution(brown, yellow) {
    const answer = [];
    
    const sum = brown + yellow;
    for (let i = parseInt(sum / 2); i > 2; i--) {
        if (sum % i === 0) {
            const num1 = i, num2 = parseInt(sum / i);
            if (num1 > 2 && num2 > 2 && (num1 - 2) * (num2 - 2) === yellow) {
                answer.push(num1);
                answer.push(num2);
                break;
            }
        }
    }
    
    return answer;
}

File: Programmers/Lv.2/카펫/42842.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int sum = brown + yellow;
        for (int i = sum; i > 2; i--) {
            if (sum % i == 0) {
                int num1 = i;
                int num2 = sum / i;
                
                if (num1 > 2 && num2 > 2 && (num1 - 2) * (num2 - 2) == yellow) {
                    answer[0] = num1;
                    answer[1] = num2;
                    break;
                }
            }
        }
        
        return answer;
    }
}
